keras_hub/src/models/segformer/segformer_image_segmenter.py

```diff
@@ -0,0 +1,194 @@
+            a dictionary with keys "P2", "P3", "P4", and "P5" and layer names as values.
+        num_classes: int, the number of classes for the detection model,
+            including the background class.
+        projection_filters: int, number of filters in the
+            convolution layer projecting the concatenated features into
+            a segmentation map. Defaults to 256`.
+
+
+    Example:
+
+    Using presets:
+
+    ```python
+    import keras_hub
+    import numpy as np
+
+    segmenter = keras_hub.models.SegFormerImageSegmenter.from_preset("segformer_b0_ade20k_512")
+
+    images = np.random.rand(1, 512, 512, 3)
+    segformer(images)
+    ```
+
+    Using the SegFormer backbone:
+
+    ```python
+    encoder = keras_hub.models.MiTBackbone.from_preset("mit_b0_ade20k_512")
+    backbone = keras_hub.models.SegFormerBackbone(image_encoder=encoder, projection_filters=256)
+    ```
+
+    Using the SegFormer backbone with a custom encoder:
+
+    ```python
+    import keras
+    import keras_hub
+    import numpy as np
+
+    images = np.ones(shape=(1, 96, 96, 3))
+    labels = np.zeros(shape=(1, 96, 96, 1))
+
+    encoder = keras_hub.models.MiTBackbone(
+        depths=[2, 2, 2, 2],
+        image_shape=(96, 96, 3),
+        hidden_dims=[32, 64, 160, 256],
+        num_layers=4,
+        blockwise_num_heads=[1, 2, 5, 8],
+        blockwise_sr_ratios=[8, 4, 2, 1],
+        max_drop_path_rate=0.1,
+        patch_sizes=[7, 3, 3, 3],
+        strides=[4, 2, 2, 2],
+    )
+
+    backbone = keras_hub.models.SegFormerBackbone(image_encoder=encoder, projection_filters=256)
+    segformer = keras_hub.models.SegFormerImageSegmenter(backbone=backbone, num_classes=4)
+
+    segformer(images)
+    ```
+
+    Using the segmentor class with a preset backbone:
+
+    ```python
+    import keras_hub
+
+    image_encoder = keras_hub.models.MiTBackbone.from_preset("mit_b0_ade20k_512")
+    backbone = keras_hub.models.SegFormerBackbone(image_encoder=encoder, projection_filters=256)
+    segformer = keras_hub.models.SegFormerImageSegmenter(backbone=backbone, num_classes=4)
+    ```
+    """
+
+    backbone_cls = SegFormerBackbone
+    preprocessor_cls = SegFormerImageSegmenterPreprocessor
+
+    def __init__(
+        self,
+        backbone,
+        num_classes,
+        preprocessor=None,
+        **kwargs,
+    ):
+        if not isinstance(backbone, keras.layers.Layer) or not isinstance(
+            backbone, keras.Model
+        ):
+            raise ValueError(
+                "Argument `backbone` must be a `keras.layers.Layer` instance "
+                f" or `keras.Model`. Received instead "
+                f"backbone={backbone} (of type {type(backbone)})."
+            )
+
+        # === Layers ===
+        inputs = backbone.input
+
+        self.backbone = backbone
+        self.preprocessor = preprocessor
+        self.dropout = keras.layers.Dropout(0.1)
+        self.output_segmentation = keras.layers.Conv2D(
```

Comment:
should we call this the `output_segmentation_head`

Reply:
👍